coderd/organizations.go

```diff
@@ -118,6 +118,98 @@ func (api *API) postOrganizations(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusCreated, convertOrganization(organization))
 }
 
+// @Summary Update organization
+// @ID update-organization
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Organizations
+// @Param organization path string true "Organization ID or name"
+// @Param request body codersdk.UpdateOrganizationRequest true "Patch organization request"
+// @Success 200 {object} codersdk.Organization
+// @Router /organizations/{organization} [patch]
+func (api *API) patchOrganization(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	organization := httpmw.OrganizationParam(r)
+
+	var req codersdk.UpdateOrganizationRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	if req.Name != organization.Name {
+		if req.Name == codersdk.DefaultOrganization && !organization.IsDefault {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: fmt.Sprintf("Organization name %q is reserved.", codersdk.DefaultOrganization),
+			})
+			return
+		}
+
+		_, err := api.Database.GetOrganizationByName(ctx, req.Name)
+		if err == nil {
+			httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
+				Message: "Organization already exists with that name.",
+			})
+			return
+		}
+		if !errors.Is(err, sql.ErrNoRows) {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: fmt.Sprintf("Internal error fetching organization %q.", req.Name),
+				Detail:  err.Error(),
+			})
+			return
+		}
```

Comment:
Remove this, since you might not be able to fetch the organization anyway.

Reply:
Not being able to fetch the org is the happy path. This is to check if the new name is already in use, and is taken basically directly from the `postOrganization` handler. I imagine we could remove this and the table constraints would just cause the update to error, but I think I prefer this for the better error messages that it'll give the frontend.

Reply:
oh I see, you moved this handling in your other suggestion